src/Components/Server/src/Builder/RazorComponentsEndpointBuilderExtensions.cs

```diff
@@ -0,0 +1,130 @@
+        /// <returns>The <paramref name="builder"/>.</returns>
+        public static RazorComponentsEndpointBuilder AddComponent<TComponent>(this RazorComponentsEndpointBuilder builder, string selector)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (selector == null)
+            {
+                throw new ArgumentNullException(nameof(selector));
+            }
+
+            return AddComponent(builder, typeof(TComponent), selector);
+        }
+
+        /// <summary>
+        /// Adds <paramref name="componentType"/> to the list of components registered with this <see cref="ComponentsHub"/> instance.
+        /// The DOM selector associated with the <paramref name="componentType"/> will default to the component type name in lowercase.
+        /// </summary>
+        /// <param name="builder">The <see cref="RazorComponentsEndpointBuilder"/>.</param>
+        /// <param name="componentType">The component type.</param>
+        /// <returns>The <paramref name="builder"/>.</returns>
+        public static RazorComponentsEndpointBuilder AddComponent(this RazorComponentsEndpointBuilder builder, Type componentType)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (componentType == null)
+            {
+                throw new ArgumentNullException(nameof(componentType));
+            }
+
+            return builder.AddComponent(componentType, componentType.Name.ToLowerInvariant());
+        }
+
+        /// <summary>
+        /// Adds <paramref name="componentType"/> to the list of components registered with this <see cref="ComponentsHub"/> instance.
+        /// The selector will default to the component name in lowercase.
+        /// </summary>
+        /// <param name="builder">The <see cref="RazorComponentsEndpointBuilder"/>.</param>
+        /// <param name="componentType">The component type.</param>
+        /// <param name="selector">The component selector in the DOM for the <paramref name="componentType"/>.</param>
+        /// <returns>The <paramref name="builder"/>.</returns>
+        public static RazorComponentsEndpointBuilder AddComponent(this RazorComponentsEndpointBuilder builder, Type componentType, string selector)
+        {
+            if (builder == null)
+            {
+                throw new ArgumentNullException(nameof(builder));
+            }
+
+            if (componentType == null)
+            {
+                throw new ArgumentNullException(nameof(componentType));
+            }
+
+            if (selector == null)
+            {
+                throw new ArgumentNullException(nameof(selector));
+            }
+
+            builder.Add(endpointBuilder => AddComponent(endpointBuilder.Metadata, componentType, selector));
+            return builder;
+        }
+
+
```

Comment:
Hmm should I comment on this extra line? Nah, I'll leave it for someone else to complain about.